new-lamassu-admin/src/pages/Customers/Customers.js

```diff
@@ -27,26 +53,126 @@ const GET_CUSTOMERS = gql`
+  const [variables, setVariables] = useState({})
+  const [filters, setFilters] = useState([])
+
+  const {
+    data: customersResponse,
+    loading: customerLoading,
+    refetch
+  } = useQuery(GET_CUSTOMERS, {
+    variables,
+    onCompleted: data => setFilteredCustomers(R.path(['customers'])(data))
+  })
+
+  const { data: filtersResponse, loading: loadingFilters } = useQuery(
+    GET_CUSTOMER_FILTERS
+  )
+
   const configData = R.path(['config'])(customersResponse) ?? []
   const locale = configData && fromNamespace(namespaces.LOCALE, configData)
   const customersData = R.sortWith([R.descend(R.prop('lastActive'))])(
-    R.path(['customers'])(customersResponse) ?? []
+    filteredCustomers ?? []
   )
 
+  const onFilterChange = filters => {
+    const filtersObject = R.compose(
+      R.mergeAll,
+      R.map(f => ({
+        [f.type]: f.value
+      }))
+    )(filters)
+
+    setFilters(filters)
+
+    setVariables({
+      phone: filtersObject.phone,
+      name: filtersObject.name,
+      address: filtersObject.address,
+      id: filtersObject.id
+    })
+
+    console.log({
+      phone: filtersObject.phone,
+      name: filtersObject.name,
+      address: filtersObject.address,
+      id: filtersObject.id
+    })
+
+    refetch && refetch()
+  }
+
+  const onFilterDelete = filter =>
+    setFilters(
+      R.filter(f => !R.whereEq(R.pick(['type', 'value'], f), filter))(filters)
+    )
+
+  const filterOptions = R.path(['customerFilters'])(filtersResponse)
+
   return (
-    <CustomersList
-      data={customersData}
-      locale={locale}
-      onClick={handleCustomerClicked}
-      loading={loading}
-    />
+    <>
+      <TitleSection
+        title="Customers"
+        appendix={
+          <div>
+            <SearchBox
+              loading={loadingFilters}
+              filters={filters}
+              options={filterOptions}
+              inputPlaceholder={'Search customers'}
+              onChange={onFilterChange}
+            />
+          </div>
+        }
+        appendixClassName={baseStyles.buttonsWrapper}
+        labels={[
+          { label: 'Cash-in', icon: <TxInIcon /> },
+          { label: 'Cash-out', icon: <TxOutIcon /> }
+        ]}
+      />
+      {filters.length > 0 && (
+        <>
+          <P className={classes.text}>{'Filters:'}</P>
+          <div>
+            {filters.map((f, idx) => (
+              <Chip
+                key={idx}
+                classes={chipClasses}
+                label={`${f.type}: ${f.value}`}
+                onDelete={() => onFilterDelete(f)}
+                deleteIcon={<CloseIcon className={classes.button} />}
+              />
+            ))}
+            <Chip
+              classes={chipClasses}
+              label={`Delete filters`}
+              onDelete={() => setFilters([])}
+              deleteIcon={<CloseIcon className={classes.button} />}
+            />
+          </div>
+        </>
+      )}
```

Comment:
Isn't going to be the same for every place we add this component? We should have something like a `SearchFilterRederer` instead of repeating this everywhere.